samples/reactjs-todo/client/components/journey/identity-provider.js

```diff
@@ -0,0 +1,67 @@
+/*
+ * forgerock-sample-web-react
+ *
+ * identity-provider.js
+ *
+ * Copyright (c) 2021 ForgeRock. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+import React from 'react';
+import { DEBUGGER } from '../../constants';
+import GoogleIcon from '../icons/google-icon';
+import AppleIcon from '../icons/apple-icon';
+/**
+ * @function IdentityProvider - IdentityProvider React component
+ * @param {Object} props - React props object passed from parent
+ * @param {Object} props.callback - The callback object from AM
+ * @returns {Object} - React component object
+ */
+export default function IdentityProvider({ callback }) {
+  const providersClass = {
+    Google: 'identity-provider google-login',
+    Apple: 'identity-provider apple-login',
+    Facebook: 'identity-provider facebook-login',
+  };
+  const identityProviders = callback
+    .getProviders()
+    .filter((provider) => provider.provider !== 'localAuthentication');
+
+  function onSetProvider(provider) {
+    if (DEBUGGER) debugger;
+    callback.setProvider(provider);
+  }
+
+  function getDisplayName(idp) {
+    switch (idp.uiConfig.buttonDisplayName) {
+      case 'Google':
+        return 'Google';
+      case 'Facebook':
+        return 'Facebook';
+      case 'Apple':
+        return 'Apple';
+      default:
+        return 'Google';
+    }
+  }
+
+  return identityProviders.map((provider, idx) => {
+    const providerName = getDisplayName(provider);
+    return (
+      <div>
+        <button
+          className={providersClass[provider.uiConfig.buttonDisplayName]}
+          key={idx}
+          onClick={() => onSetProvider(provider.provider)}
+        >
+          {providerName === 'Google' ? (
+            <GoogleIcon size="24px" />
+          ) : providerName === 'Apple' ? (
+            <AppleIcon size="24px" />
+          ) : null}{' '}
```

Comment:
Is there a reason we have Facebook in the logic above, but not implemented here?

Reply:
we've had the same question for the angular implementation. I did it this way because the idp service didn't come along with a facebook icon. Just followed the design